result/result.py

```diff
@@ -272,3 +294,48 @@ def result(self) -> Result[Any, Any]:
         Returns the original result.
         """
         return self._result
+
+
+def as_result(
+    *exceptions: Type[TBE],
+) -> Callable[[Callable[P, R]], Callable[P, Result[R, TBE]]]:  # type: ignore[misc]
+    """
+    Make a decorator to turn a function into one that returns a ``Result``.
+
+    Regular return values are turned into ``Ok(return_value)``. Raised
+    exceptions of the specified exception type(s) are turned into ``Err(exc)``.
+    """
+    # Note: type annotations for signature-preserving decorators via ParamSpec
+    # are currently not supported by Mypy 0.910; see
+    # https://github.com/python/mypy/issues/8645
+    #
+    # The ‘type: ignore’ comments below are for our own linting purposes.
+    # Calling code works without errors from Mypy, but will also not be
+    # type-safe, i.e. it will behave as if it is calling a function like
+    # f(*args: Any, **kwargs: Any)
+    if not exceptions or not all(
+        inspect.isclass(exception) and issubclass(exception, BaseException)
+        for exception in exceptions
+    ):
+        raise TypeError("as_result() requires one or more exception types")
+
+    def decorator(
+        f: Callable[P, R],  # type: ignore[misc]
+    ) -> Callable[P, Result[R, TBE]]:  # type: ignore[misc]
+        """
+        Decorator to turn a function into one that returns a ``Result``.
+        """
+
+        @functools.wraps(f)
+        def wrapper(
+            *args: P.args,  # type: ignore[name-defined]
+            **kwargs: P.kwargs,  # type: ignore[name-defined]
+        ) -> Result[R, TBE]:
+            try:
+                return Ok(f(*args, **kwargs))
+            except exceptions as exc:
```

Comment:
exceptions is a list no? Is this valid syntax to pass in a list to the `except` statement?

Reply:
I'm mistaken. `*args` is a tuple, not a list, and I was playing around with syntax in repl using a list, `[IndexError, TypeError]`. A tuple is valid, so this is correct 👍

Reply:
yes, the `except ...` statement takes an arbitrary expression, and a tuple (not a list) of exception classes is the right way to catch more than one type. details at https://docs.python.org/3/reference/compound_stmts.html#the-try-statement